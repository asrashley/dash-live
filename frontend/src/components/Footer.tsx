import { useRef } from "preact/hooks";

declare const _GIT_HASH_: string;

export function Footer() {
    const year = useRef<number>(new Date().getFullYear());
    return <footer className="bg-body-tertiary footer row border-top border-secondary-subtle">
    <div className="col-3 github-link">
      <div><a href="https://github.com/asrashley/dash-live">github.com/asrashley/dash-live</a></div>
      <div>{_GIT_HASH_}</div>
    </div>
    <div className="col-6 text-center service-name">Simulated MPEG DASH service</div>
    <div className="col-3 text-end copyright">&copy;{year.current} Alexis Ashley</div>
  </footer>;

}