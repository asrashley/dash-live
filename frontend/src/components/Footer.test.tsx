import { afterAll, beforeAll, describe, expect, test, vi } from "vitest";
import { renderWithProviders } from "../test/renderWithProviders";
import { Footer } from "./Footer";

describe('Footer component', () => {
    beforeAll(() => {
        vi.useFakeTimers();
        vi.setSystemTime(new Date('2025-01-01T00:00:00Z'));
    });

    afterAll(() => {
        vi.restoreAllMocks();
        vi.useRealTimers();
    });

    test('matches snapshot', () => {
        const { asFragment, getByText } = renderWithProviders(<Footer />);
        getByText(new Date().getFullYear().toString(), { exact: false });
        getByText(window["_GIT_HASH_"]);
        expect(asFragment()).toMatchSnapshot();
    });
});