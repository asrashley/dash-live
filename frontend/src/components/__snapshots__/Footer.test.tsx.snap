// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer component > matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="bg-body-tertiary footer row border-top border-secondary-subtle"
  >
    <div
      class="col-3 github-link"
    >
      <div>
        <a
          href="https://github.com/asrashley/dash-live"
        >
          github.com/asrashley/dash-live
        </a>
      </div>
      <div>
        12345abcde
      </div>
    </div>
    <div
      class="col-6 text-center service-name"
    >
      Simulated MPEG DASH service
    </div>
    <div
      class="col-3 text-end copyright"
    >
      Â©2025 Alexis Ashley
    </div>
  </footer>
</DocumentFragment>
`;
